core/src/clients/snyk/snyk_client.rs

```diff
@@ -0,0 +1,223 @@
+use std::{collections::HashMap, fs, fmt::format};
+
+use async_trait::async_trait;
+use platform::hyper::{ContentType, Error as HyperError, get, post};
+use serde_json::Value;
+use crate::{clients::{OrgProjects, SnykProviderError}, config::get_snyk_access_token};
+
+use super::{SnykData, Org};
+
+#[cfg(test)]
+use mockall::{automock, mock, predicate::*};
+
+//URL used for getting org and project details
+const API_V1_URL: &'static str = "https://snyk.io/api/v1";
+//URL used for getting SBOMS
+const API_V3_URL: &'static str = "https://api.snyk.io/rest";
+const API_V3_VERSION: &'static str = "2023-02-15~beta";
+//Format of SBOMs returned from Snyk
+const SBOM_FORMAT: &'static str = "cyclonedx%2Bjson";
+
+//TODO: Finish unit tests
+///Snyk API trait implementation
+pub struct SnykRestClientImpl{}
```

Comment:
I think idiomatic rust or go would just call this `Client`. We already know it's for Snyk because of the module name. Not a blocker, especially since we don't have a naming conventions document, but would like to stick with common naming conventions so I figured I'd share.

Reply:
Ah yeah that makes sense, I will rename that